handlebars/handlebars.go

```diff
@@ -198,11 +199,15 @@ func (e *Engine) Render(out io.Writer, template string, binding interface{}, lay
 		if lay == nil {
 			return fmt.Errorf("render: layout %s does not exist", layout[0])
 		}
-		var bind map[string]interface{}
-		if m, ok := binding.(map[string]interface{}); ok {
-			bind = m
-		} else {
-			bind = make(map[string]interface{}, 1)
+		bind := make(map[string]interface{})
```

Comment:
is it really good performance technically to always create one map and loop the data into the other map when we could just take the other map instead ?

i think the cause for the bugs was only the type check, which you can map via reflect

Reply:
Something like this?

	var bind map[string]interface{}
	if m, ok := binding.(fiber.Map); ok {
		bind = m
	} else {
		bind = make(map[string]interface{}, 1)
	}

Reply:
Which removes the need for reflection as well.

Reply:
This is perhaps too rigid
at least you can use the template system without fiber

what is the fix if we customize it like you mentioned in your comment? because the fiber.Map is the same as the map[string]interface{}. 

https://github.com/gofiber/fiber/blob/87cedc2022cc3ebfe3ca4ec5621154feb9a313e5/app.go#L37

Reply:
https://go.dev/play/p/ukiwgq8XPMu

we should use such a check
```go
reflect.ValueOf(binding).Kind() == reflect.Map
```

Reply:
This is an example of the bug:
https://go.dev/play/p/TKsZIR0T5az

We need to convert the `binding` variable from type `interface{}` to type `map[string]interface{}`. This is to add a new index in the map that contains the parsed view string to embed in the layout. I think this can only be done by assertion from the original datatype `fiber.map` or if we want to keep it generic I think we need to unwind the datatype and assign it to a new available of type `map[string]interface{}`?

Reply:
Here is an example of supporting both `fiber.Map` and a generic `map[string]interface{}`:
https://go.dev/play/p/uQ0szykw8fQ